web_src/js/features/pull-view-file.js

```diff
@@ -0,0 +1,66 @@
+import {setFileFolding} from './file-fold.js';
+
+const prReview = window.config.pageData.prReview || {};
+const viewedStyleClass = 'viewed-file-checked-form';
+const viewedCheckboxSelector = '.viewed-file-form'; // Selector under which all "Viewed" checkbox forms can be found
+
+// Refreshes the summary of viewed files if present
+// The data used will be window.config.pageData.prReview.numberOf{Viewed}Files
+function refreshViewedFilesSummary() {
+  const viewedFilesMeter = document.getElementById('viewed-files-summary');
+  if (viewedFilesMeter) viewedFilesMeter.setAttribute('value', prReview.numberOfViewedFiles);
```

Comment:
Could be me, but I haven't seen this before in this codebase, a `if` with a single statement. Anyhow this case can be simplified:
```suggestion
  viewedFilesMeter && viewedFilesMeter.setAttribute('value', prReview.numberOfViewedFiles);
```

Reply:
Well, with jQuery, it can be as simple as `$viewdFilesMeter.val(prReview.numberOfViewedFiles)`

Reply:
I don't think it justify to import jQuery, to only simplify one expression here.

Reply:
I mean, in most cases for go-tmpl and Fomantic UI, jQuery is the best choice.

Reply:
@Gusted surprisingly, I haven't seen your syntax anywhere else in the source files. My quick search with
ag '&&' | grep -v 'if' shows a few results but all of which are used to set or return a boolean.
None of them conditionally sets attributes.
On the other side, ag 'if\s*\(.*;' shows a few more results, however, most of which are used to return prematurely.
I'd rather leave it at the current state than your suggestion as I'd say the current approach is clearer while yours is smaller, and JS is a language where clarity is more important than conciseness.

---

@wxiaoguang I'm trying not to use JQuery wherever possible.
I definitely support https://github.com/go-gitea/gitea/issues/18345#issuecomment-1017733409 in his attempt to throw out JQuery completely.

Reply:
Why not just use:
`viewedFilesMeter?.setAttribute('value', prReview.numberOfViewedFiles);`?

Reply:
Is that syntax already supported?
Wasn't that only added with ES2021?

Reply:
Okay, I tested it. Seems to be supported already.
Caniuse says that 90% of browsers support it, so it should be fine.
Will change it with the next commit.
Only the call two lines below must continue to exist as that cannot be simplified.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining It's not that _new_, but I'm not sure about the specifics of the minimum browsers so we might want to avoid it for now.



> @Gusted surprisingly, I haven't seen your syntax anywhere else in the source files. My quick search with
> `ag '&&' | grep -v 'if'` shows a few results but all of which are used to set or return a boolean.
> None of them conditionally sets attributes.
> On the other hand, `ag 'if\s*\(.*;'` shows a few more results, however, most of which are used to return prematurely.
> I'd rather leave it at the current state than your suggestion as I'd say the current approach is clearer while yours is smaller, and JS is a language where clarity is more important than conciseness.

Yeah, quick and dirty javascript shenanigans :) Javascript is full of such one-liners 😄. It's a matter of taste, I'm personally more used to only see if conditions for more complex situations